node/tessel-export.js

```diff
@@ -169,6 +230,15 @@ Tessel.Port = function(name, socketPath, board) {
   // Array of {size, callback} used to dispatch replies
   this.replyQueue = [];
 
+  // When the push function is called
+  this.replyQueue.push = function () {
+    // Make sure this socket is referenced until the data is returned
+    this.sock._ref();
+
+    // Push the data into the array
+    return Array.prototype.push.apply(this,arguments);
```

Comment:
`this` isn't an array, it's the `port` instance that was bound on the line immediately following. The result is that nothing gets pushed into `this.replyQueue`. Even if that's fixed, this approach also doesn't account for `this.replyQueue.shift()`. I recommend strictly avoiding the redefinition of built-in object's methods.

Alternative: 

```js
Tessel.Port.prototype.enqueue = function(reply) {
  this.ref();
  this.replyQueue.push(reply);
};


Tessel.Port.prototype.dequeue = function() {
  this.unref();
  return this.replyQueue.shift();
};
```

- all calls to `this.replyQueue.push(...)` become `this.enqueue(...)`
- all calls to `this.replyQueue.shift()` become `this.dequeue()`